packages/std-client/src/setup/setupMUDV2Network.ts

```diff
@@ -0,0 +1,196 @@
+    },
+    {
+      id: "LoadingState",
+      metadata: { contractId: "component.LoadingState" },
+    }
+  );
+
+  const storeSchemaTableId = new TableId("mudstore", "schema");
+  const storeSchemaComponent = defineComponent(
+    world,
+    { schema: Type.T },
+    {
+      metadata: {
+        contractId: storeSchemaTableId.toHexString(),
+        tableId: storeSchemaTableId.toString(),
+      },
+    }
+  );
+
+  const storeComponents = defineStoreComponents(world, storeMudConfig);
+  const worldComponents = defineStoreComponents(world, worldMudConfig);
+  const userComponents = defineStoreComponents(world, options.mudConfig);
+
+  const components = {
+    // v2 components
+    storeSchemaComponent,
+    ...storeComponents,
+    ...worldComponents,
+    ...userComponents,
+    // v1 components
+    SystemsRegistry,
+    ComponentsRegistry,
+    LoadingState,
+  } satisfies Components;
+
+  storeComponents.StoreMetadata;
+
+  // Mapping from component contract id to key in components object
+  const mappings: Mappings<typeof components> = {};
+
+  // Function to register new components in mappings object
+  function registerComponent(key: string, component: ContractComponent) {
+    const { contractId, tableId } = component.metadata;
+    if (tableId) {
+      mappings[tableId] = key;
+    } else {
+      mappings[keccak256(contractId)] = key;
+    }
+  }
+
+  // Register initial components in mappings object
+  for (const key of Object.keys(components)) {
+    registerComponent(key, components[key]);
+  }
+
+  const network = await createNetwork(options.networkConfig);
+  world.registerDisposer(network.dispose);
+
+  const signerOrProvider = computed(() => network.signer.get() || network.providers.get().json);
+
+  const { contracts, config: contractsConfig } = await createContracts<{ World: WorldContract }>({
+    config: { World: { abi: WorldAbi, address: options.networkConfig.worldAddress } },
+    signerOrProvider,
+  });
+
+  const gasPriceInput$ = new BehaviorSubject<number>(
+    // If no initial gas price is provided, check the gas price once and add a 30% buffer
+    options?.initialGasPrice || Math.ceil((await signerOrProvider.get().getGasPrice()).toNumber() * 1.3)
+  );
+
+  const { txQueue, dispose: disposeTxQueue } = createTxQueue(contracts, network, gasPriceInput$, {
+    devMode: options.networkConfig.devMode,
+  });
+  world.registerDisposer(disposeTxQueue);
+
+  const { systems, registerSystem, getSystemContract } = createSystemExecutor<SystemTypes>(
+    world,
+    network,
+    SystemsRegistry,
+    SystemAbis,
+    gasPriceInput$,
+    {
+      devMode: options.networkConfig.devMode,
+    }
+  );
+
+  const decodeNetworkComponentUpdate = createDecodeNetworkComponentUpdate(world, components, mappings);
+  const { systemCallStreams, decodeAndEmitSystemCall } = createSystemCallStreams(
+    world,
+    keys(SystemAbis),
+    SystemsRegistry,
+    getSystemContract,
+    decodeNetworkComponentUpdate
+  );
```

Comment:
same question here: wondering if we can just replace this with the separate v2 call streams approach to avoid having to make it compatible with v1 types, etc.

Reply:
yeah I think that's reasonable